CAP/gap/CategoryConstructor.gi

```diff
@@ -37,6 +37,7 @@ InstallMethod( CategoryConstructor,
         create_func_object_or_fail := IsObject,
         create_func_morphism := IsObject,
         create_func_morphism_or_fail := IsObject,
+        create_func_list_of_objects := IsList,
```

Comment:
```suggestion
        create_func_list_of_objects := IsObject,
```

This checks the type of the options. The option `create_func_list_of_objects` can be either a function or the string "default", so we generically check `IsObject`.